beetmoverscript/src/beetmoverscript/script.py

```diff
@@ -303,23 +308,69 @@ async def async_main(context):
 
     validate_task_schema(context)
 
+    # determine the task bucket and action
+    #   Note: bucket here is the config key under bucket_config, for aws/gcs
+    #         platform "bucket" name use get_bucket_name()
+    context.bucket = get_task_bucket(context.task, context.config)
```

Comment:
While you're in here refactoring, you may want to keep in #171, which would get us to stop using the `context` object from scriptworker, and start passing around dictionaries directly. Definitely scope creep, but it's definitely more of a bug than a feature that we import scriptworker in our scriptworker-scripts.

Reply:
I think we should fix this in a different PR. It touches pretty much every function in beetmoverscript.